### ISSUE TITLE
Add and Retrieve Custom Tags from REST API

### ISSUE THREAD

**Issue by chintzs** [dadoonet/fscrawler#607]
Hi,

I am using fscrawler v2.4. It has been working exceptionally well for me and my requirement. Recently I've had a requirement to add custom tags using the REST API when the file is being added to the crawler. I've tried using the following code
`curl -F "file=@sharedocs.pdf" -F "tags=@tags.txt" "http://127.0.0.1:8080/fscrawler/_upload"`
my tags.txt is as below
`{
  "content": "OVERWRITE CONTENT",
  "external": {
    "name": "sharedocs",
    "id": 34,
    "description": "these are additional tags"
  }
}`
However, I'm not able to search based on these tags.

Please let me know if any more information I can post as fscrawler is working fine for me for all the other tasks which needs to be performed and have never required to post any issues.

**Comment by dadoonet** [comment]
What is a typical document that has been generated?
What search are you running?

**Comment by dadoonet** [comment]
That'd be easier if you could format your code and make it more readable.

Anyway, I can see that in the final document there is no `external.name` field. That's why you can't search for it.

So I guess that this did not work:

```sh
curl -F "file=@sharedocs.pdf" -F "tags=@tags.txt" "http://127.0.0.1:8080/fscrawler/_upload"
```

I started to test it but I hit another bug (#608) which is kind of blocking me for now.
I'll update this when I'll be able to test.

**Comment by dadoonet** [comment]
@chintzs So I just tried with the latest 2.6-SNAPSHOT:

Configuration file: `~/.fscrawler/rest/_settings.json`

```json
{
  "name" : "rest",
  "fs" : {
    "url" : "/tmp"
  },
  "rest" : {
    "scheme" : "HTTP",
    "host" : "127.0.0.1",
    "port" : 8080,
    "endpoint" : "fscrawler"
  }
}
```

Start FSCrawler:

```sh
bin/fscrawler rest --loop 0 --rest --debug
```

Generate `test.txt` and `tags.txt`:

```sh
echo "This is my text" > test.txt
echo "{\"content\":\"OVERWRITE CONTENT\",\"external\":{\"tenantId\": 23,\"projectId\": 34,\"description\":\"these are additional tags\"}}" > tags.txt
```

Send it to FSCrawler:

```sh
$ curl -F "file=@test.txt" -F "tags=@tags.txt" "http://127.0.0.1:8080/fscrawler/_upload"
```

Response is:

```json
{
  "ok" : true,
  "filename" : "test.txt",
  "url" : "http://127.0.0.1:9200/rest/_doc/dd18bf3a8ea2a3e53e2661c7fb53534",
  "doc" : {
    "content" : "OVERWRITE CONTENT",
    "meta" : { },
    "file" : {
      "extension" : "txt",
      "content_type" : "text/plain; charset=ISO-8859-1",
      "indexing_date" : "2018-09-24T07:39:31.757+0000",
      "filename" : "test.txt"
    },
    "path" : {
      "virtual" : "test.txt",
      "real" : "test.txt"
    },
    "external" : {
      "tenantId" : 23,
      "projectId" : 34,
      "description" : "these are additional tags"
    }
  }
}
```

Checking in Elasticsearch:

```sh
$ curl http://127.0.0.1:9200/rest/_doc/dd18bf3a8ea2a3e53e2661c7fb53534?pretty
```

Gives

```json
{
  "_index" : "rest",
  "_type" : "_doc",
  "_id" : "dd18bf3a8ea2a3e53e2661c7fb53534",
  "_version" : 1,
  "found" : true,
  "_source" : {
    "content" : "OVERWRITE CONTENT",
    "meta" : { },
    "file" : {
      "extension" : "txt",
      "content_type" : "text/plain; charset=ISO-8859-1",
      "indexing_date" : "2018-09-24T07:39:31.757+0000",
      "filename" : "test.txt"
    },
    "path" : {
      "virtual" : "test.txt",
      "real" : "test.txt"
    },
    "external" : {
      "tenantId" : 23,
      "projectId" : 34,
      "description" : "these are additional tags"
    }
  }
}
```

So I don't see what your problem is. May be share the document you used?
If you could share the exact steps you did as I just did, that might help to reproduce the problem.
Could you also test with FSCrawler 2.5 at least? So many things have been fixed after 2.4.

I'm closing but feel free to add new comments or reopen if you can reproduce the same problem with 2.5.
